src/bundler.zig

```diff
@@ -1536,8 +1536,12 @@ pub const Bundler = struct {
             },
             // TODO: use lazy export AST
             .text => {
+                const size = std.mem.replacementSize(u8, source.contents, "\\", "\\\\");
```

Comment:
do we need to worry about freeing source.contents? I think the answer is no because they should be allocated in the threadlocal heap. But I'm not 100% sure.

Reply:
Might be a memory leak, since running `bun-debug test regression` with this change results in a failed assertion at the end of the tests:
```
...
 23 pass
 0 fail
 snapshots: +1 added
 83 expect() calls
Ran 23 tests across 18 files. [2.63s]
mimalloc: assertion failed: at "/Users/zng/projects/bun/src/deps/mimalloc/src/alloc.c":246, mi_page_usable_size_of
  assertion: "ok"
Abort trap: 6
```

Using the previous commit and running regression test (with the new tests checked out) results in:
```
...
1 tests failed:
✗ importing .txt files should not parse backslashes [22.03ms]

 22 pass
 1 fail
 snapshots: +1 added
 82 expect() calls
Ran 23 tests across 18 files. [2.67s]
``` with no failed assertions.

However the weird thing is that running `bun-debug run 03449-fixture.ts` with these changes and `bun-debug test 03449.test.ts` doesn't cause an assertion fail, and executes properly.

Reply:
This means the allocator in use should be  bun.default_allocator

Reply:
@Jarred-Sumner I've updated the PR with deallocation through the `allocator` local var. I tried using `bun.default_allocator`, but that still caused a mimalloc assertion to fail, but using `allocator` allows the test to pass without any failed assertions.